qiskit/transpiler/passes/utils/gate_direction.py

```diff
@@ -95,6 +97,90 @@ def _rzx_dag(parameter):
+            if isinstance(node.op, ControlFlowOp):
+                continue
+            qargs = (wire_map[node.qargs[0]], wire_map[node.qargs[1]])
+            if qargs not in edges and (qargs[1], qargs[0]) not in edges:
+                raise TranspilerError(
+                    f"The circuit requires a connection between physical qubits {qargs}"
+                )
+            if qargs not in edges:
+                replacement = self._static_replacements.get(node.name)
+                if replacement is not None:
+                    dag.substitute_node_with_dag(node, replacement)
+                elif node.name == "rzx":
+                    dag.substitute_node_with_dag(node, self._rzx_dag(*node.op.params))
+                else:
+                    raise TranspilerError(
+                        f"Flipping of gate direction is only supported "
+                        f"for {list(self._static_replacements)} at this time, not '{node.name}'."
+                    )
+        for node in dag.op_nodes(ControlFlowOp):
+            node.op = node.op.replace_blocks(
+                dag_to_circuit(
+                    self._run_coupling_map(
+                        circuit_to_dag(block),
+                        {inner: wire_map[outer] for outer, inner in zip(node.qargs, block.qubits)},
+                        edges,
+                    )
+                )
+                for block in node.op.blocks
+            )
+        return dag
+
+    def _run_target(self, dag, wire_map):
+        for node in dag.two_qubit_ops():
+            if isinstance(node.op, ControlFlowOp):
+                continue
+            qargs = (wire_map[node.qargs[0]], wire_map[node.qargs[1]])
+            swapped = (qargs[1], qargs[0])
+            if node.name in self._static_replacements:
+                if self.target.instruction_supported(node.name, qargs):
+                    continue
+                if self.target.instruction_supported(node.name, swapped):
+                    dag.substitute_node_with_dag(node, self._static_replacements[node.name])
+                else:
+                    raise TranspilerError(
+                        f"The circuit requires a connection between physical qubits {qargs}"
+                        f" for {node.name}"
+                    )
+            elif node.name == "rzx":
+                if self.target.instruction_supported(
+                    qargs=qargs, operation_class=RZXGate, parameters=node.op.params
+                ):
+                    continue
+                if self.target.instruction_supported(
+                    qargs=swapped, operation_class=RZXGate, parameters=node.op.params
+                ):
+                    dag.substitute_node_with_dag(node, self._rzx_dag(*node.op.params))
+                else:
+                    raise TranspilerError(
+                        f"The circuit requires a connection between physical qubits {qargs}"
+                        f" for {node.name}"
```

Comment:
Oh yeah - since `Target.instruction_supported` got added, the todo became very easy to action.  I think we're probably leaving some efficiency on the table here when it's called in a loop with `operation_class` set, but that's for another day most likely.